### PR TITLE
Upgrade to Chrome 44

### PR DESCRIPTION
**Pull request by zcbenz**
* [x] Not building on OS X
* [x] Not building on Linux
* [x] Not building on Windows
* [x] Renderer process crashes when using Buffer
* [x] Renderer process crashes when printing Buffer in devtools
* [x] Rewrite the protocol module (#2475)
* [x] Crash when using protocol module
* [x] Warn about old protocol API usages
* [x] Make release build working
* [x] Make native modules work
* [x] Pass tests
* [x] Merge https://github.com/atom/brightray/pull/127

### CODE REVIEW

File: atom/browser/net/js_asker.cc [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+  else
+    state->Set(called_symbol, v8::Boolean::New(isolate, true));
+
+  // If there is no argument passed then we failed.
+  scoped_ptr<CallbackHolder> holder(
+      static_cast<CallbackHolder*>(external->Value()));
+  CHECK(holder);
+  v8::Local<v8::Value> value;
+  if (!args->GetNext(&value)) {
+    holder->callback.Run(false, nullptr);
```
Comment (deepak1556) [inline atom/browser/net/js_asker.cc]: @zcbenz shouldnt the failure also be notified on the IO thread ?

Comment (zcbenz) [inline atom/browser/net/js_asker.cc]: Didn't notice that, thanks!

**Comment by hokein** [comment]
:clap: :sparkles:

**Comment by paulcbetts** [comment]
:clap:

**Comment by frankhale** [comment]
Thank you @hokein and @zcbenz for your valiant effort.You guys are super hero's to us mortals!

**Pull request merged**